x-pack/plugin/sql/qa/src/main/java/org/elasticsearch/xpack/sql/qa/jdbc/PreparedStatementTestCase.java

```diff
@@ -207,4 +301,28 @@ public void testSingleParameterMultipleTypes() throws Exception {
             return result;
         }
     }
+
+    private static long convertFromCalendarToUTC(long value, Calendar cal) {
+        if (cal == null) {
+            return value;
+        }
+        Calendar c = (Calendar) cal.clone();
+        c.setTimeInMillis(value);
+
+        ZonedDateTime convertedDateTime = ZonedDateTime
+            .ofInstant(c.toInstant(), c.getTimeZone().toZoneId())
+            .withZoneSameLocal(ZoneOffset.UTC);
+
+        return convertedDateTime.toInstant().toEpochMilli();
+    }
+
+    private static long testMillis(long randomMillis, int i) {
+        long millis = randomMillis;
+        if (i == 1) {
+            millis -= 1;
+        } else if (i == 3) {
+            millis += 1;
+        }
+        return millis;
```

Comment:
Just a more concise alternative suggestion: `return randomMillis - 2 + i` (tho not perfectly equivalent).